### ISSUE TITLE
Add "keep private" to enable drafts of longterm projects

### ISSUE THREAD

**Issue by lauradereynal** [mozilla/webmaker-core#651]
From a user : 
- I am working on it, but for now I havent published it. *

It's hard to understand what is public and what is not. I felt embarrassed that she thought her project was private, but it could have been featured in the gallery before she was ready. 

@vazquez @xmatthewx

**Comment by vazquez** [comment]
Big point here. It also speaks to our minimal sense of activity (some work on that happening in #640). We have talked briefly and in passing about drafts, but this definitely needs more discussion.

**Comment by xmatthewx** [comment]
There are a few different solutions across a spectrum: make things more clear or give the user more control. More control can introduce a lot of UX complexity. There's ambiguity between save, publish, and share.  

I propose we draft a design for consideration. Give users a "hide // show" toggle on each project in the make view. Let's validate this and other ideas before we invest too much energy. 

The best solution might just be more clarity.

**Comment by xmatthewx** [comment]
Worth noting that this is an issue because most projects are rich content that take time to build (unlike a quick status or photo post). The app rides a blurry line between these two paradigms.

**Comment by vazquez** [comment]
#### Some explorations

We can place a visibility toggle in different parts of the app.

##### Make View
- Content would be in draft mode/invisible by default. The UI would subtly reflect this.
- Upon marking a content as visible, the Android Snackbar would reveal to show a confirmation message.

<img width="598" alt="screenshot 2015-09-17 10 51 46" src="https://cloud.githubusercontent.com/assets/1119821/9936779/a7534eec-5d2b-11e5-9279-31f410b37bc7.png">

---

##### Editor View
- It might make more sense to place the visibility toggle on the Editor View since it's where the user might decide if a project is ready to be discoverable.
- Snackbar appears upon toggle.

<img width="679" alt="screenshot 2015-09-17 12 52 18" src="https://cloud.githubusercontent.com/assets/1119821/9940008/27031438-5d3b-11e5-81e5-4a67d0dcdbfe.png">

---

##### Editor View FTU
- Placing visibility toggle on the Editor also gives us an opportunity to educate through FTU.
- We can toggle the following when a user presses the back button for the first time:

<img width="352" alt="screenshot 2015-09-17 12 57 57" src="https://cloud.githubusercontent.com/assets/1119821/9940105/c1551810-5d3b-11e5-8826-4abac5f818c4.png">

---

Thought about placing switchers in the header, but I still have to work this more.

<img width="344" alt="screenshot 2015-09-17 13 01 13" src="https://cloud.githubusercontent.com/assets/1119821/9965183/83763a48-5e02-11e5-8c9d-c11f11c1b347.png">

**Comment by xmatthewx** [comment]
Nice exploration. A few thoughts:
- This fits well in Make view. Provides a good overview of all your projects and their visibility. We would want a snackbar to display on toggle to state 'your project is now visible' or 'your project is now private'.
- In the editor, it draws attention away from more important functionality. It gives a lot of prominence to something you might utilize only once in the lifetime of a project. 
- Another option: settings view. Adding a title is the moment an author is thinking about how others will perceive the project. This space would handle a toggle with a nice clear label.

**Comment by vazquez** [comment]
Quick clarification: The snackbar is already visible in the top Make View explorations, were you thinking something in addition to that?

Make view would be the ideal spot if we made projects not-visible by default. Settings would be great if projects were visible by default. The advantage to having projects not-visible by default is that we could gain a higher quality of content.

---

### Updating Make View

![screenshot 2015-09-29 11 56 50](https://cloud.githubusercontent.com/assets/1119821/10169904/2e04197a-66a1-11e5-9d6a-50a4a7f81ff2.png)

---

### Settings View Exploration

We would need to remove the ability to inline edit the Title and Description fields (and move them to modal edits), as switches are used in conjunction with labels, not text fields.

![screenshot 2015-09-29 11 55 52](https://cloud.githubusercontent.com/assets/1119821/10169869/0ee26ce0-66a1-11e5-984b-30afc2cc9c90.png)

**Comment by vazquez** [comment]
During our standup, we talked about adding a CTA on the Make View if the user has not added any metadata. This could help us in:
1. Encouraging unique Titles
2. Descriptions with hashtags in the future!
3. Show the user that the visibility toggle appears in Settings

The `Name This Project` button would take the user to the **Settings View**.

![scan sep 29 2015 1 37 pm](https://cloud.githubusercontent.com/assets/1119821/10173078/7a61ebc6-66b0-11e5-8eed-6219e20be520.png)

**Comment by xmatthewx** [comment]
:+1: on the button.

**Comment by xmatthewx** [comment]
We'll need to be careful with visibility settings to avoid ambiguity about privacy or suggest that flipping the switch will definitely add things as a feature to Discover. As currently worded, many people might want to share their project to friends, but still keep the switch off. We should also avoid complex tech requirements that require auth in the browser viewer or require us to change previously shared URLs.

> **Keep this project private**
> Do not show it on my profile or help others find it via Discover. (Any links you shared with other people will still work.)

Given the potential ambiguity, I don't think we should offer a toggle in the Make view. We could include the eye icon as an indicator of the settings.

**Comment by vazquez** [comment]
Understood, and great points. New Project Settings would look like the following:

![image](https://cloud.githubusercontent.com/assets/1119821/10290589/ff1fb2c8-6b72-11e5-88c6-8df62527d37c.png)

Before we rule out the eye icon in the Make View, what about this exploration? The button would not act as a toggle, but if the user were to tap on the icon, we could offer help.

![image](https://cloud.githubusercontent.com/assets/1119821/10290839/6b3c89e4-6b74-11e5-8dc7-4fae4d585ba8.png)

**Comment by xmatthewx** [comment]
The privacy icon is good, and might work better next to the title. Or, in it's current location, it could extend the touch target on the kabob instead of showing the tooltip. Kabob, of course, should reveal a menu that includes "project details". 

Let's file the "name this project" CTA as a separate issue, and create and wrap up the spec for this privacy setting.

**Comment by vazquez** [comment]
:+1: On it

**Comment by vazquez** [comment]
Let me know what you think @xmatthewx and I will file a separate implementation issue for cleanliness.

![page background annotations copy](https://cloud.githubusercontent.com/assets/1119821/10670351/22b811ec-78b3-11e5-835d-b044e1a059b2.png)

**Comment by vazquez** [comment]
Filed #729 for the Name Your Project

**Comment by xmatthewx** [comment]
r+ :+1:

**Comment by vazquez** [comment]
![post project](https://cloud.githubusercontent.com/assets/1119821/11882217/3d80b98c-a4d9-11e5-8099-ef3c7fc5be73.png)

**Comment by xmatthewx** [comment]
Beautiful work. I filed this as a new clean ticket so we can close this monster. #935
